lib/rules/sort-invocations.js

```diff
@@ -0,0 +1,315 @@
+
+          if (!isValueUndefined) {
+            return b.attr(name, b.mustache(b.string('')));
+          }
+        }
+
+        break;
+      }
+    }
+
+    return b.attr(name, value);
+  });
+}
+
+function sortHash(hash) {
+  return b.hash(hash.pairs.sort(compareHashPairs));
+}
+
+function sortModifiers(modifiers) {
+  return modifiers.sort(compareModifiers).map((modifier) => {
+    const { hash, params, path } = modifier;
+
+    return b.elementModifier(path, params, hash);
+  });
+}
+
+export default class SortInvocations extends Rule {
+  visitor() {
+    return {
+      BlockStatement(node) {
+        const { hash, params, path } = node;
+
+        if (canSkipSortHash(hash)) {
+          return;
+        }
+
+        const sorted = sortHash(hash);
+
+        if (this.mode === 'fix') {
+          node = b.block(path, params, sorted, b.blockItself());
+        } else {
+          this.log({
+            isFixable: true,
+            message: `Hash keys must be sorted: ${sorted.pairs.map(getHashPairName).join(', ')}`,
+            node,
+          });
+        }
+      },
+
+      ElementNode(node) {
+        const { attributes, modifiers } = node;
+
+        if (!canSkipSortAttributes(attributes)) {
+          const sorted = sortAttributes(attributes);
+
+          if (this.mode === 'fix') {
+            node.attributes = sorted;
+          } else {
+            this.log({
+              isFixable: true,
+              message: `Arguments and attributes must be sorted: ${sorted.map(getAttributeName).join(', ')}`,
+              node,
+            });
+          }
+        }
+
+        if (!canSkipSortModifiers(modifiers)) {
+          const sorted = sortModifiers(modifiers);
+
+          if (this.mode === 'fix') {
+            node.modifiers = sorted;
+          } else {
+            this.log({
+              isFixable: true,
+              message: `Modifiers must be sorted: ${sorted.map(getModifierName).join(', ')}`,
+              node,
+            });
+          }
+        }
+      },
+
+      MustacheStatement(node) {
+        const { hash } = node;
+
+        if (canSkipSortHash(hash)) {
+          return;
+        }
+
+        const sorted = sortHash(hash);
+
+        if (this.mode === 'fix') {
+          node.hash = sorted;
+        } else {
+          this.log({
+            isFixable: true,
+            message: `Hash keys must be sorted: ${sorted.pairs.map(getHashPairName).join(', ')}`,
+            node,
+          });
+        }
+      },
+
+      SubExpression(node) {
+        const { hash, params, path } = node;
+
+        if (canSkipSortHash(hash)) {
+          return;
+        }
+
+        const sorted = sortHash(hash);
+
+        if (this.mode === 'fix') {
+          node = b.sexpr(path, params, sorted);
+        } else {
+          this.log({
+            isFixable: true,
+            message: `Hash keys must be sorted: ${sorted.pairs.map(getHashPairName).join(', ')}`,
+            node,
```

Comment:
I think we may scary user with whole `node` in RED_UNDERLINE (in editor), I'm proposing to log fist incorrectly sorted node here.